client/clip_client/client.py

```diff
@@ -109,19 +111,21 @@ def encode(self, content, **kwargs):
             not kwargs.get('show_progress'),
             total=len(content) if hasattr(content, '__len__') else None,
         )
+        results = DocumentArray()
         with self._pbar:
             self._client.post(
-                **self._get_post_payload(content, kwargs), on_done=self._gather_result
+                **self._get_post_payload(content, kwargs),
+                on_done=partial(self._gather_result, results=results),
             )
-        return self._unboxed_result
+        return self._unboxed_result(results, self._return_plain)
```

Comment:
`self._return_plain` suffer from the same issue as `self._results`

Reply:
Good point